### PR TITLE
Add support for transmission rpc_whitelist settings, quell chef_gem compile_time warnings

### PR DESCRIPTION
**Pull request by elijah**
Minor updates - this adds the ability to toggle on and off the IP whitelist features, as well as setting the whitelist if one wants.

I also have quelled the warnings for chef_gem by setting compile_time to true.  [The transmission-simple gem seems to be what breaks, here - but since we have to build *it* anyway - it didn't make sense to break the other gems out into a different %w package block.  Someone could choose to do that, but... I don't think it matters much at this time.]

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @elijah

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

File: recipes/default.rb [hunk 1]
```diff
@@ -23,6 +23,7 @@
 %w(bencode i18n transmission-simple activesupport).each do |pkg|
   chef_gem pkg do
     action :install
+    compile_time true
```
Comment (tas50) [inline recipes/default.rb]: This needs to be guarded so we don't break older versions of chef that lacked this attribute.

compile_time true if Chef::Resource::ChefGem.method_defined?(:compile_time)

Comment (elijah) [inline recipes/default.rb]: Fixed by adding conditional as requested.

**Comment by tas50** [comment]
Thanks @elijah

**Comment by elijah** [comment]
Thanks for the merge, Tim!!

--e

**Pull request merged**